### ISSUE TITLE
Panel / Bokeh based app and performance

### ISSUE THREAD

**Issue by willirath** [ESM-VFC/ESM-VFC#6]
(@suvarchal this follows up on the discussion about panel-based apps we had via email.)

I've played with panel today and found it to execute the whole notebook each time a new client appears / a new session is created. I'm don't know for sure if this is intended behaviour, but I find it very likely, because always each session from scratch is the only way to be sure that sessions don't interfere.

I could achieve considerable speedup if I did all loading and evaluation in Dask and hacked the notebook to share the scheduler (first session started dask on a given port and further sessions re-used this if it was already running).

**Comment by suvarchal** [comment]
Right, is see the need that has to redo it every session. so to minimize re execution speed, we need to have make all code in notebook as functions as much as possible and minimize any printing. But that makes it less of a notebook. So in the end notebook serving two roles as a notebook and as an app  has its downsides.

excited to see your dasky hacks.

**Comment by willirath** [comment]
@suvarchal Here's a short demo that re-discovers a Dask cluster (knowing host and port) in subsequent sessions: https://github.com/willirath/panel-with-shared-dask-cluster

My impression is that this speeds up multiple sessions with lots of data handling by a lot.
